c-layer/lib.cxx

```diff
@@ -86,35 +92,43 @@ static zypp::ZYpp::Ptr zypp_ptr() {
   return NULL;
 }
 
-// TODO: split init target into set of repo manager, initialize target and load target and merge it in rust
-void init_target(const char *root, struct Status *status, ProgressCallback progress, void *user_data) noexcept {
+// TODO: split init target into set of repo manager, initialize target and load
+// target and merge it in rust
+struct Zypp *init_target(const char *root, struct Status *status,
+                         ProgressCallback progress, void *user_data) noexcept {
   const std::string root_str(root);
 
+  struct Zypp *zypp = NULL;
   try {
     zypp::RepoManagerOptions repo_manager_options(root);
-    // repository manager options cannot be replaced, a new repository manager is needed
-    zypp::RepoManager *new_repo_manager = new zypp::RepoManager(repo_manager_options);
+    // repository manager options cannot be replaced, a new repository manager
+    // is needed
+    zypp::RepoManager *new_repo_manager =
+        new zypp::RepoManager(repo_manager_options);
 
     // replace the old repository manager
-    if (repo_manager)
-      delete repo_manager;
-    repo_manager = new_repo_manager;
+    if (the_zypp.repo_manager)
+      delete the_zypp.repo_manager;
```

Comment:
I think we can make code more robust if we check here that the_zypp was previously correctly free. So this call should not happen as
```
zypp1 = init_target("/");
zypp2 = init_target("/mnt");
// using zypp1 and zypp2 here
free_zypp(zypp1);
free_zypp(zypp2);
```
probably won't do what you expect ( unless zypp is really context ready, which I suspend is not the case ).